pubsublite-spark-sql-streaming/src/main/java/com/google/cloud/pubsublite/spark/PslPartitionOffset.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.pubsublite.spark;
+
+import com.google.auto.value.AutoValue;
+import com.google.cloud.pubsublite.Offset;
+import com.google.cloud.pubsublite.Partition;
+import com.google.cloud.pubsublite.SubscriptionPath;
+import java.io.Serializable;
+import org.apache.spark.sql.sources.v2.reader.streaming.PartitionOffset;
+
+@AutoValue
+public abstract class PslPartitionOffset implements PartitionOffset, Serializable {
```

Comment:
same question, why is this serializable? Isn't it an in-memory only datastructure?

Reply:
This is serialized and transported between driver and executor(worker).